### ISSUE TITLE
Examples of correct code for no-unneeded-ternary

### ISSUE THREAD

**Issue by Mouvedia** [eslint/eslint#8211]
This is a discussion/question.

Why are
```
var a = x ? y : x;

var a = x ? x : 1;
```
considered correct?

cf http://eslint.org/docs/rules/no-unneeded-ternary#rule-details

You could write them as
```
var a = x && y;

var a = x || 1;
```

What am I missing?

> This rule disallow ternary operators when simpler alternatives exist.

**Comment by platinumazure** [comment]
Those examples do indeed seem correct.

Worth noting that this would not be the case if either of the variables were actually MemberExpressions. In that case, if the member in question has a getter function, then switching away from the ternary would potentially result in one fewer invocation of the getter.

**Comment by not-an-aardvark** [comment]
I agree, it would be nice if the rule caught these.

Other cases:

```js
!x ? y : x;
!x ? x : y;

// gets fixed to ->

x || y;
x && y;
```

I think the rule should still report MemberExpressions, but not fix them. Getters with side-effects are very rare, so the report would still be useful in the vast majority of cases.

**Comment by kaicataldo** [comment]
Agreed that the rule should check this. Looks like we need an implementation to move this forward, though. Since we're working on v4, we could make this a breaking change that the rule reports by default.

**Comment by Mouvedia** [comment]
And maybe a new `sideEffect` option set to false by default.

Anyhow as a first step not fixing but just reporting—as @not-an-aardvark proposed—will do for potential/detected side-effects.

**Comment by not-an-aardvark** [comment]
Looking into this, it seems like the first cases are already partially covered by the `defaultAssignment` option, but the option isn't very powerful (it only checks identifiers at the moment).

**Comment by nzakas** [comment]
Unfortunately, it looks like there wasn't enough interest from the team or community to implement this change. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to be implemented after 90 days tend to never be implemented, and as such, we [close those issues](https://eslint.org/docs/maintainer-guide/issues#when-to-close-an-issue). This doesn't mean the idea isn't interesting or useful, just that it's not something the team can commit to.

**Comment by Mouvedia** [comment]
Seems like an oversight to me: should be fixed.
